### PR TITLE
[Improvement](ES)Supprt datav2 and datetimev2 for es query

### LINKED ISSUE #16645: [Bug] Malformed packet error will query on es table
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master

### What's Wrong?

image

When using a `select * limit 10` sql to query es table, I got these issue.

### What You Expected?

Get right result

### How to Reproduce?

I create a es table with the new version of DataTimeV2 and DecimalV3,  and these may cuase these problem since BE had not impl these two type.

### Anything Else?
These configuration should be set in properties in create table sql:
```
"max_docvalue_fields"="120"
```

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by SaintBacchus**
# Proposed changes

Issue Number: close #16645

## Problem summary
Now doris support query only date and datetime type for es query.

Make it support the new datev2 and datetimev2 type too.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [x] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [x] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.44 seconds
 stream load tsv:          475 seconds loaded 74807831229 Bytes, about 150 MB/s
 stream load json:         39 seconds loaded 2358488459 Bytes, about 57 MB/s
 stream load orc:          68 seconds loaded 1101869774 Bytes, about 15 MB/s
 stream load parquet:          25 seconds loaded 861443392 Bytes, about 32 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230211071321_clickbench_pr_94381.html

File: be/src/exec/es/es_scroll_parser.cpp [hunk 1]
```diff
@@ -528,72 +530,104 @@ Status ScrollParser::fill_columns(const TupleDescriptor* tuple_desc,
     return Status::OK();
 }
 
-Status ScrollParser::fill_date_slot_with_strval(void* slot, const rapidjson::Value& col,
```
Comment (SaintBacchus) [inline be/src/exec/es/es_scroll_parser.cpp:531]: this function was not used

**Comment by stalary** [comment]
Can you add some regression-test case?

File: be/src/exec/es/es_scroll_parser.cpp [hunk 2]
```diff
@@ -528,72 +530,104 @@ Status ScrollParser::fill_columns(const TupleDescriptor* tuple_desc,
...
+        if (type == TYPE_DATE) {
+            dt_val.cast_to_date();
+        } else {
+            dt_val.to_datetime();
+        }
+
+        auto date_packed_int = binary_cast<doris::vectorized::VecDateTimeValue, int64_t>(
+                *reinterpret_cast<vectorized::VecDateTimeValue*>(&dt_val));
+        col_ptr->insert_data(const_cast<const char*>(reinterpret_cast<char*>(&date_packed_int)), 0);
+        return Status::OK();
```
Comment (stalary) [inline be/src/exec/es/es_scroll_parser.cpp:552]: `col_ptr->insert_data(const_cast<const char*>(reinterpret_cast<char*>(&date_packed_int)), 0);
        return Status::OK();` can move to out of if and else

File: be/src/exec/es/es_scroll_parser.cpp [hunk 3]
```diff
@@ -528,72 +530,104 @@ Status ScrollParser::fill_columns(const TupleDescriptor* tuple_desc,
...
+        } else {
+            reinterpret_cast<vectorized::VecDateTimeValue*>(&dt_val)->set_type(TIME_DATETIME);
+        }
 
-    auto date_packed_int = binary_cast<doris::vectorized::VecDateTimeValue, int64_t>(
-            *reinterpret_cast<vectorized::VecDateTimeValue*>(&dt_val));
-    col_ptr->insert_data(const_cast<const char*>(reinterpret_cast<char*>(&date_packed_int)), 0);
+        auto date_packed_int = binary_cast<doris::vectorized::VecDateTimeValue, int64_t>(
+                *reinterpret_cast<vectorized::VecDateTimeValue*>(&dt_val));
+        col_ptr->insert_data(const_cast<const char*>(reinterpret_cast<char*>(&date_packed_int)), 0);
```
Comment (stalary) [inline be/src/exec/es/es_scroll_parser.cpp:599]: Same as above

File: be/src/exec/es/es_scroll_parser.cpp (hunk 2 above)
Comment (SaintBacchus) [reply be/src/exec/es/es_scroll_parser.cpp:553]: date_packed_int is different type

**Review by stalary — APPROVED** [review]
LGTM

**Pull request merged**